lib/ClangImporter/ImportDecl.cpp

```diff
@@ -9094,6 +9096,47 @@ class SwiftifyInfoPrinter {
 };
 } // namespace
 
+namespace {
+  /// Look for any side effects within a Stmt.
+  struct CATExprValidator : clang::ConstStmtVisitor<CATExprValidator, bool> {
+    bool VisitDeclRefExpr(const clang::DeclRefExpr *e) { return true; }
+    bool VisitIntegerLiteral(const clang::IntegerLiteral *) { return true; }
+    bool VisitImplicitCastExpr(const clang::ImplicitCastExpr *c) { return this->Visit(c->getSubExpr()); }
+    bool VisitParenExpr(const clang::ParenExpr *p) { return this->Visit(p->getSubExpr()); }
+
+#define SUPPORTED_UNOP(UNOP) \
+    bool VisitUnary ## UNOP(const clang::UnaryOperator *unop) { \
+      return this->Visit(unop->getSubExpr()); \
+    }
+    SUPPORTED_UNOP(Plus)
+    SUPPORTED_UNOP(Minus)
+    SUPPORTED_UNOP(Not)
+#undef SUPPORTED_UNOP
+
+#define SUPPORTED_BINOP(BINOP) \
+    bool VisitBin ## BINOP(const clang::BinaryOperator *binop) { \
+      return this->Visit(binop->getLHS()) && this->Visit(binop->getRHS()); \
+    }
+    SUPPORTED_BINOP(Add)
+    SUPPORTED_BINOP(Sub)
+    SUPPORTED_BINOP(Div)
+    SUPPORTED_BINOP(Mul)
+    SUPPORTED_BINOP(Rem)
+    SUPPORTED_BINOP(Shl)
+    SUPPORTED_BINOP(Shr)
+    SUPPORTED_BINOP(And)
+    SUPPORTED_BINOP(Xor)
+    SUPPORTED_BINOP(Or)
+#undef SUPPORTED_BINOP
+
+    bool VisitStmt(const clang::Stmt *) { return false; }
+  };
+} // namespace
+
+static bool SwiftifiableCAT(const clang::CountAttributedType *CAT) {
```

Comment:
Could we get some more descriptive names here? I see what "CAT" is an acronym for here, but it isn't quite discoverable. `CountedByExpressionValidator`, perhaps?

Reply:
Yeah that's a fair point! I'll address that in my PR on main.